### ISSUE TITLE
LSP spec violation: sending RenameOptions when prepareSupport is not set

### ISSUE THREAD

**Issue by thomasjm** [castwide/solargraph#207]
I noticed that `solargraph` seems to be returning `RenameOptions` for its `renameProvider` setting when it returns capabilities, even though I haven't set the `prepareSupport` flag. This seems to be a violation of the spec, which says the following:

```
	/**
	 * The server provides rename support. RenameOptions may only be
	 * specified if the client states that it supports
	 * `prepareSupport` in its initial `initialize` request.
	 */
	renameProvider?: boolean | RenameOptions;
```

(It's causing my LSP client to fail to parse the `renameProvider`.)

**Comment by thomasjm** [comment]
CC @mattn , this seems to have been introduced in #158

**Comment by castwide** [comment]
There's a fix for this in the `master` branch. It'll be included in the next patch release.

**Comment by castwide** [comment]
Released in v0.34.3.

**Comment by thomasjm** [comment]
Thanks!
